content/client-lib-development-guide/features.textile

```diff
@@ -107,6 +107,7 @@ h3(#rest-auth). Auth
 *** @(RSA4c1)@An @ErrorInfo@ with @code@ @80019@ and description of the underlying failure should be emitted with the state change, in the @errorReason@ and/or in the callback as appropriate
 *** @(RSA4c2)@If the connection is @CONNECTING@, then the connection attempt should be treated as unsuccessful, and as such the connection should transition to the @DISCONNECTED@ or @SUSPENDED@ state as defined in "RTN14":#RTN14 and "RTN15":#RTN15
 *** @(RSA4c3)@If the connection is @CONNECTED@, then the connection should remain @CONNECTED@
+** @(RSA4d)@ If an attempt to authenticate is made using the @authUrl@ and the content type is neither text nor JSON, or if using @authCallback@ and the object passed in is neither a @String@, @JsonObject@, @TokenRequest@ object nor a @TokenDetails@ object, the auth attempt should fail with error code @40143@ and a suitable error message.  Additionally, if the text token string is greater than 384 bytes, or the JSON stringified @JsonObject@, @TokenRequest@ or @TokenDetails@ is greater than 128kb, then the auth attempt should fail with error code @40143@ and a suitable error message.
```

Comment:
> Because of a problem we've seen recently whereby a 500kb payload was sent to Ably

I follow that, I just meant that in all of those problems are random html getting treated as a token string, not random json getting treated as a token details, because that shouldn't happen - it wouldn't get detected as one. So not sure that's needed. But shrug, I don't mind it if you think it's needed, so sure.

> The spec item is not describing token failure conditions, it is instead describing side effects of a failure.

True, but we want it to be clear that 4c side-effects *do* apply to the failure in this PR. The way it's phrased in this PR makes it seem like an unrelated requirement. Especially as your new 4d also describes side-effects of the failure, which kinda clash with 4c.

So eg if you wanted to keep this at 4d, you could modify 4c to say "the provided token is in an invalid format **(per RSA4d)**, or ...", then have `(RSA4d) The following conditions imply that a token is in an invalid format: the content-type of the response from @authUrl@ is neither @text/plain@ or @application/json@; the object returned by @authCallback@ is not a @String@, @TokenRequest@ object, or @TokenDetails@ object; or if the text token string is greater than 384 bytes, or the JSON stringified @JsonObject@, @TokenRequest@ or @TokenDetails@ is greater than 128kb.`